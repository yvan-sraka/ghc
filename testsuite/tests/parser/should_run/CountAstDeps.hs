module Main(main) where

-- Calculate the number of module dependencies of 'Parser.' If that
-- number exceeds a threshold, that indicates that the dependencies
-- have significantly gone up via the commit under test (and the test
-- is deemed to fail). In that case, this most likely means a cycle
-- has arisen that pulls in modules for Core generation. The
-- motivation for not allowing that to happen is so that the
-- 'ghc-lib-parser' package subset of the GHC API can continue to be
-- provided with as small a number of modules as possible for when the
-- need exists to produce ASTs and nothing more.

import CountDeps

main :: IO ()
main = printDeps "Language.Haskell.Syntax"
