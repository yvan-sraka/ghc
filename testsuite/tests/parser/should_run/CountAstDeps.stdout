Found 238 Language.Haskell.Syntax module dependencies
GHC.Builtin.Names
GHC.Builtin.PrimOps
GHC.Builtin.Types
GHC.Builtin.Types.Prim
GHC.Builtin.Uniques
GHC.ByteCode.Types
GHC.Cmm
GHC.Cmm.BlockId
GHC.Cmm.CLabel
GHC.Cmm.Dataflow.Block
GHC.Cmm.Dataflow.Collections
GHC.Cmm.Dataflow.Graph
GHC.Cmm.Dataflow.Label
GHC.Cmm.Expr
GHC.Cmm.MachOp
GHC.Cmm.Node
GHC.Cmm.Switch
GHC.Cmm.Type
GHC.CmmToAsm.CFG.Weight
GHC.CmmToAsm.Config
GHC.Core
GHC.Core.Class
GHC.Core.Coercion
GHC.Core.Coercion.Axiom
GHC.Core.Coercion.Opt
GHC.Core.ConLike
GHC.Core.DataCon
GHC.Core.FVs
GHC.Core.FamInstEnv
GHC.Core.InstEnv
GHC.Core.Lint
GHC.Core.Make
GHC.Core.Map.Type
GHC.Core.Multiplicity
GHC.Core.Opt.Arity
GHC.Core.Opt.CallerCC
GHC.Core.Opt.ConstantFold
GHC.Core.Opt.Monad
GHC.Core.Opt.OccurAnal
GHC.Core.PatSyn
GHC.Core.Ppr
GHC.Core.Predicate
GHC.Core.Seq
GHC.Core.SimpleOpt
GHC.Core.Stats
GHC.Core.Subst
GHC.Core.TyCo.FVs
GHC.Core.TyCo.Ppr
GHC.Core.TyCo.Rep
GHC.Core.TyCo.Subst
GHC.Core.TyCo.Tidy
GHC.Core.TyCon
GHC.Core.TyCon.Env
GHC.Core.TyCon.RecWalk
GHC.Core.Type
GHC.Core.Unfold
GHC.Core.Unfold.Make
GHC.Core.Unify
GHC.Core.UsageEnv
GHC.Core.Utils
GHC.CoreToIface
GHC.Data.Bag
GHC.Data.BooleanFormula
GHC.Data.EnumSet
GHC.Data.FastMutInt
GHC.Data.FastString
GHC.Data.FastString.Env
GHC.Data.FiniteMap
GHC.Data.Graph.Directed
GHC.Data.IOEnv
GHC.Data.List.SetOps
GHC.Data.Maybe
GHC.Data.OrdList
GHC.Data.Pair
GHC.Data.Stream
GHC.Data.StringBuffer
GHC.Data.TrieMap
GHC.Driver.Backend
GHC.Driver.CmdLine
GHC.Driver.Env
GHC.Driver.Env.Types
GHC.Driver.Errors
GHC.Driver.Flags
GHC.Driver.Hooks
GHC.Driver.Monad
GHC.Driver.Phases
GHC.Driver.Pipeline.Monad
GHC.Driver.Plugins
GHC.Driver.Ppr
GHC.Driver.Session
GHC.Hs
GHC.Hs.Binds
GHC.Hs.Decls
GHC.Hs.Doc
GHC.Hs.Expr
GHC.Hs.Extension.GhcPass
GHC.Hs.ImpExp
GHC.Hs.Instances
GHC.Hs.Lit
GHC.Hs.Pat
GHC.Hs.Type
GHC.Hs.Utils
GHC.Iface.Ext.Fields
GHC.Iface.Recomp.Binary
GHC.Iface.Syntax
GHC.Iface.Type
GHC.Linker.Types
GHC.Parser.Annotation
GHC.Platform
GHC.Platform.AArch64
GHC.Platform.ARM
GHC.Platform.Constants
GHC.Platform.NoRegs
GHC.Platform.PPC
GHC.Platform.Profile
GHC.Platform.Reg
GHC.Platform.Reg.Class
GHC.Platform.Regs
GHC.Platform.S390X
GHC.Platform.SPARC
GHC.Platform.Ways
GHC.Platform.X86
GHC.Platform.X86_64
GHC.Prelude
GHC.Runtime.Context
GHC.Runtime.Eval.Types
GHC.Runtime.Heap.Layout
GHC.Runtime.Interpreter.Types
GHC.Settings
GHC.Settings.Config
GHC.Settings.Constants
GHC.Stg.Syntax
GHC.StgToCmm.Types
GHC.SysTools.BaseDir
GHC.SysTools.FileCleanup
GHC.SysTools.Terminal
GHC.Tc.Errors.Hole.FitTypes
GHC.Tc.Types
GHC.Tc.Types.Constraint
GHC.Tc.Types.Evidence
GHC.Tc.Types.Origin
GHC.Tc.Utils.TcType
GHC.Types.Annotations
GHC.Types.Avail
GHC.Types.Basic
GHC.Types.CompleteMatch
GHC.Types.CostCentre
GHC.Types.CostCentre.State
GHC.Types.Cpr
GHC.Types.Demand
GHC.Types.Error
GHC.Types.FieldLabel
GHC.Types.Fixity
GHC.Types.Fixity.Env
GHC.Types.ForeignCall
GHC.Types.ForeignStubs
GHC.Types.HpcInfo
GHC.Types.Id
GHC.Types.Id.Info
GHC.Types.Id.Make
GHC.Types.Literal
GHC.Types.Meta
GHC.Types.Name
GHC.Types.Name.Cache
GHC.Types.Name.Env
GHC.Types.Name.Occurrence
GHC.Types.Name.Ppr
GHC.Types.Name.Reader
GHC.Types.Name.Set
GHC.Types.RepType
GHC.Types.SafeHaskell
GHC.Types.SourceError
GHC.Types.SourceFile
GHC.Types.SourceText
GHC.Types.SrcLoc
GHC.Types.Target
GHC.Types.TyThing
GHC.Types.TypeEnv
GHC.Types.Unique
GHC.Types.Unique.DFM
GHC.Types.Unique.DSet
GHC.Types.Unique.FM
GHC.Types.Unique.Set
GHC.Types.Unique.Supply
GHC.Types.Var
GHC.Types.Var.Env
GHC.Types.Var.Set
GHC.Unit
GHC.Unit.Env
GHC.Unit.External
GHC.Unit.Finder.Types
GHC.Unit.Home
GHC.Unit.Home.ModInfo
GHC.Unit.Info
GHC.Unit.Module
GHC.Unit.Module.Deps
GHC.Unit.Module.Env
GHC.Unit.Module.Graph
GHC.Unit.Module.Imported
GHC.Unit.Module.Location
GHC.Unit.Module.ModDetails
GHC.Unit.Module.ModGuts
GHC.Unit.Module.ModIface
GHC.Unit.Module.ModSummary
GHC.Unit.Module.Name
GHC.Unit.Module.Status
GHC.Unit.Module.Warnings
GHC.Unit.Parser
GHC.Unit.Ppr
GHC.Unit.State
GHC.Unit.Types
GHC.Utils.Binary
GHC.Utils.Binary.Typeable
GHC.Utils.BufHandle
GHC.Utils.CliOption
GHC.Utils.Error
GHC.Utils.Exception
GHC.Utils.FV
GHC.Utils.Fingerprint
GHC.Utils.GlobalVars
GHC.Utils.IO.Unsafe
GHC.Utils.Json
GHC.Utils.Lexeme
GHC.Utils.Misc
GHC.Utils.Monad
GHC.Utils.Outputable
GHC.Utils.Panic
GHC.Utils.Panic.Plain
GHC.Utils.Ppr
GHC.Utils.Ppr.Colour
Language.Haskell.Syntax
Language.Haskell.Syntax.Binds
Language.Haskell.Syntax.Decls
Language.Haskell.Syntax.Expr
Language.Haskell.Syntax.Extension
Language.Haskell.Syntax.Lit
Language.Haskell.Syntax.Pat
Language.Haskell.Syntax.Type
